streamalert_cli/terraform/helpers.py

```diff
@@ -24,3 +29,33 @@ def terraform_check():
                        'your $PATH:\n'
                        '\t$ export PATH=$PATH:/usr/local/terraform/bin')
     return run_command(['terraform', 'version'], error_message=prereqs_message, quiet=True)
+
+def terraform_state_lock(action, region, table):
+    ddb_client = boto3.client('dynamodb', region_name=region)
+    ddb_tables = ddb_client.list_tables()
+    if action == 'create':
+        if table not in ddb_tables['TableNames']:
+            LOGGER.info('Creating terraform state locking DynamoDB table')
+            ddb_client.create_table(
+                AttributeDefinitions=[{
+                    'AttributeName': 'LockID',
+                    'AttributeType': 'S'
+                }],
+                TableName=table,
+                KeySchema=[{
+                    'AttributeName': 'LockID',
+                    'KeyType': 'HASH'
+                }],
+                BillingMode='PAY_PER_REQUEST'
+            )
+            wait = True
```

Comment:
there are utility functions in boto3 that implement this for you. See: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Waiter.TableExists

you don't have to implement this yourself here.

Reply:
something like:

```
ddb_client.create_table(
    AttributeDefinitions=[{
        'AttributeName': 'LockID',
        'AttributeType': 'S'
    }],
    TableName=table,
    KeySchema=[{
        'AttributeName': 'LockID',
        'KeyType': 'HASH'
    }],
    BillingMode='PAY_PER_REQUEST'
)

waiter = ddb_client.get_waiter('table_exists')
waiter.wait(TableName=table)
```

Reply:
Ohhh very good to know